test/lib/NonLinearDutchDecayLib.t.sol

```diff
@@ -10,21 +10,23 @@ import {V3DutchOutput, V3DutchInput, NonlinearDutchDecay} from "../../src/lib/V3
 import {Uint16Array, toUint256} from "../../src/types/Uint16Array.sol";
 import {ArrayBuilder} from "../util/ArrayBuilder.sol";
 import {CurveBuilder} from "../util/CurveBuilder.sol";
+import {BlockNumberish} from "../../src/base/BlockNumberish.sol";
 
 /// @notice mock contract to test NonlinearDutchDecayLib functionality
 contract MockNonlinearDutchDecayLibContract {
     function decay(
         NonlinearDutchDecay memory curve,
         uint256 startAmount,
         uint256 decayStartBlock,
+        uint256 blockNumberish,
         uint256 minAmount,
         uint256 maxAmount
     ) public view {
-        NonlinearDutchDecayLib.decay(curve, startAmount, decayStartBlock, minAmount, maxAmount);
+        NonlinearDutchDecayLib.decay(curve, startAmount, decayStartBlock, blockNumberish, minAmount, maxAmount);
     }
 }
 
-contract NonlinearDutchDecayLibTest is Test, GasSnapshot {
+contract NonlinearDutchDecayLibTest is Test, GasSnapshot, BlockNumberish {
```

Comment:
Wondering if we should do another test suite for chainId 42161 that uses `setBlockNumber` since our current tests just use `vm.roll()` which wouldn't do anything in an Arbitrum environment

Reply:
Ahh yeah so imo the NonLinearDutchDecayLib should function completely agnostic of the behavior of BlockNumberish - since it strictly takes it in as a parameter. So a test that the blockNumberish parameter passed into the Lib is correctly set may be better in the reactor test